docroot/modules/custom/mymodule/src/Form/CustomSettingsForm.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Drupal\mymodule\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Configure mymodule settings for this site.
+ */
+class CustomSettingsForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'mymodule_custom_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['mymodule.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('mymodule.settings');
+    $form['custom_name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Custom Name'),
+      '#default_value' => $config->get('custom_name'),
+    ];
+    $form['custom_checkbox'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('This is a checkbox example.'),
+      '#default_value' => $config->get('custom_checkbox'),
+    ];
+    $form['actions']['#type'] = 'actions';
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Save Custom Configs'),
+      '#button_type' => 'primary',
+    ];
```

Comment:
we dont add submit button but 
```
return parent::buildForm($form, $form_state);
```
and thus button added by parent class itself.

Reply:
Hi @joshirohit100,
I added it to see change in Button Text.
I had `parent::buildForm($form, $form_state);` in return value earlier but I wanted to change the text of submit button and calling parent was overwriting my changes.